include/dynamatic/Support/CFG.h

```diff
@@ -97,8 +97,26 @@ bool isBackedge(Value val, Operation *user, BBEndpoints *endpoints = nullptr);
 /// user (the function will assert if that is not the case).
 bool isBackedge(Value val, BBEndpoints *endpoints = nullptr);
 
-/// Determines whether the Handshake operation cannot belong to the implicit CFG
-/// of a Handshake function (i.e., cannot have a basic block attribute).
+/// Data structure to hold all edges belonging to a pair of endpoints. This
+/// constitutes an Arc. srcBB and dstBB can be equal when the arcs are
+/// Backedges. An edge is represented by means of an mlir::OpOperand.
+struct BBArc {
+  unsigned srcBB;
+  unsigned dstBB;
+  mlir::SmallVector<OpOperand *> edges;
```

Comment:
I assume that there is no meaningful notion of order between the edges, hence I don't see a rationale for this being a vector. Anything against making this a `DenseSet<OpOperand*>`? If iteration order needs to be deterministic, `SetVector<OpOperand*>` is also fine.

Reply:
I agree that `DenseSet<OpOperand*>` is a better option. The only implication of this change was the need of replacing the `push_back` method to an `insert` in `CFG.cpp`.

Reply:
Perfect, thanks for the fix.